inference-engine/src/cldnn_engine/ops/normalize_l2.cpp

```diff
@@ -33,20 +34,22 @@ void CreateNormalizeL2Op(Program& p, const std::shared_ptr<ngraph::op::v0::Norma
     }
 
     // We create fake scale constant and fill it with ones to keep the same behavior as current primitive
-    auto scale = std::make_shared<ngraph::op::v0::Constant>(op->get_output_element_type(0), ngraph::Shape{1}, std::vector<float>{1.0});
-    cldnn::layout constLayout = cldnn::layout(DataTypeFromPrecision(op->get_output_element_type(0)), cldnn::format::bfyx, cldnn::tensor{1});
-    auto mem = cldnn::memory::allocate(p.GetEngine(), constLayout, 0, false);
-    auto tmpPointer = mem.pointer<char>();  // implicitly maps buffer - unmap in destructor
-    auto buf = tmpPointer.data();
-    auto bufSize = scale->get_output_tensor(0).size();
+    if (!p.InQueryMode()) {
+        auto scale = std::make_shared<ngraph::op::v0::Constant>(op->get_output_element_type(0), ngraph::Shape{ 1 }, std::vector<float>{1.0});
+        cldnn::layout constLayout = cldnn::layout(DataTypeFromPrecision(op->get_output_element_type(0)), cldnn::format::bfyx, cldnn::tensor{ 1 });
+        auto mem = cldnn::memory::allocate(p.GetEngine(), constLayout, 0, false);
```

Comment:
As I understand, the issue happens because engine is not initialized in query mode, so p.GetEngine() returns nullptr.
Since we have other operations which requires memory allocation, I'd say that we just need to initialize engine in query mode and keep the primitive creation code as is.